src/js/angular/aclmanagement/model.js

```diff
@@ -4,94 +4,155 @@ export class ACListModel {
+        }
     }
 
-    removeRule(index) {
-        this.aclRules.splice(index, 1);
+    /**
+     * Removes a rule from the specified scope at the given index.
+     *
+     * @param {string} scope - The scope from which to remove the rule.
+     * @param {number} index - The index of the rule to remove.
+     */
+    removeRule(scope, index) {
+        const ruleList = this._aclRules.get(scope);
+        if (ruleList) {
+            ruleList.splice(index, 1);
+        }
     }
 
-    replaceRule(index, rule) {
-        this.aclRules.splice(index, 1, rule);
+    /**
+     * Replaces a rule at a specified index in the rule list for a given scope.
+     * If the scope does not exist or the rule list is empty, this method does nothing.
+     *
+     * @param {string} scope - The scope for which to replace the rule.
+     * @param {number} index - The index at which to replace the rule in the rule list.
+     * @param {object} rule - The new rule to replace the existing rule at the specified index.
+     */
+    replaceRule(scope, index, rule) {
+        const ruleList = this._aclRules.get(scope);
+        if (ruleList) {
+            ruleList.splice(index, 1, rule);
+        }
     }
 
-    getRule(index) {
-        return this.aclRules[index];
+    /**
+     * Returns the rule at the specified index for the given scope.
+     * @param {string} scope - The scope for which to retrieve the rule.
+     * @param {number} index - The index of the rule to retrieve.
+     * @returns {object|null} - The rule at the specified index for the given scope, or null if no rule exists.
+     */
+    getRule(scope, index) {
+        const ruleList = this._aclRules.get(scope);
+        return ruleList ? ruleList[index] : null;
     }
 
     /**
-     * Checks if the rule at <code>index</code> is duplicated.
-     * @param {number} indexRuleToBeChecked - the index of the rule to be checked.
-     * @return {boolean} true if the model has a duplication of the rule at <code>index</code>.
+     * Checks if the rule at the specified <code>index</code> is duplicated.
+     *
+     * @param {string} scope - The scope of the rule to be checked.
+     * @param {number} indexRuleToBeChecked - The index of the rule to be checked.
+     * @returns {boolean} - True if the model has a duplication of the rule at the specified index, otherwise false.
      */
-    isRuleDuplicated(indexRuleToBeChecked) {
-        const checkedRule = this.aclRules[indexRuleToBeChecked];
-        return this.aclRules.some((rule, index) => {
-            if (indexRuleToBeChecked !== index) {
+    isRuleDuplicated(scope, indexRuleToBeChecked) {
+        const checkedRule = this.getRule(scope, indexRuleToBeChecked);
+        if (!checkedRule) {
+            return false;
+        }
+
+        return this._aclRules.get(scope).some((rule, index) => {
+            if (index !== indexRuleToBeChecked) {
                 return angular.equals(checkedRule, rule);
```

Comment:
Get rid if this `angular.equals`. Either use lodash or something custom

Reply:
done